File: backend/models/Weight.js
const mongoose = require("mongoose");
const Schema = mongoose.Schema;

const weightSchema = new Schema({
    current_weight: {
      type: String,
      required: true,
    },
    updated_at: {
      type: Date,
      default: Date.now,
    },
    created_at: {
      type: Date,
      default: Date.now,
    },
    created_by: {
      type: Schema.Types.ObjectId,
      ref: "User",
    },
    updated_by: {
      type: Schema.Types.ObjectId,
      ref: "User",
    },
  },
  { timestamps: true }
);

module.exports = mongoose.model("Weight", weightSchema);

File: doshex_weight_tracker/lib/models/weight.dart
class Weight {
  final String current_weight;
  final String created_at;
  // final String? updatedAt;

  const Weight({required this.current_weight, required this.created_at});

  factory Weight.fromJson(Map<String, dynamic> json) {
    return Weight(
      current_weight: json['current_weight'],
      created_at: json['created_at'],
    );
  }
}
